Fixes #26173 - added config_report_metric_count metric

lzap opened the pull request:
Adds a new metric - count of status report metrics, e.g. failed,
applied, restarted etc.

This will help to correlate with other monitoring data.

To test, enable monitoring and upload a puppet report. This should work
with any report basically if plugin provides the report metric
capability.

lzap committed:
Fixes #26173 - added config_report_metric_count metric

tbrisker commented on app/services/config_report_status_calculator.rb:
```diff
@@ -59,6 +59,10 @@ def pending?
   end
 
+  def to_h
+    Hash[ConfigReport::METRIC.map {|metric| [metric, status_of(metric)] }]
```
Isn't this exactly the same result as the `status` method returns?

lzap replied:
Yup! Very good you passed reviewer mk2 exam test!

lzap commented:
Rebased, this time without a useless method.

tbrisker commented on app/services/report_importer.rb:
```diff
@@ -150,6 +150,11 @@ def create_report_and_logs
     scan
 
+    # Report metric counts via telemetry
+    ConfigReportStatusCalculator.new(:bit_field => status).status.each do |metric, count|
```
Would this work for report importers that aren't ConfigReportImporters but inherit from this class, such as omaha? https://github.com/theforeman/foreman_omaha/blob/0a5abeb5a54d95cedb92170887a65327d00eebf2/app/services/foreman_omaha/omaha_report_importer.rb#L20 
cc: @timogoebel

timogoebel replied:
I think this won't work. Good catch. Can we simply move this code into `ConfigReportImporter`?

lzap replied:
Good catch however that does not stop us from merging this small improvement, does it? ;)

timogoebel replied:
Let's move this to `ConfigReportImporter` or it'll break plugins.

```ruby
class ConfigReportImporter < ReportImporter
   def create_report_and_logs
      super

      # your code here
    end
end
```

lzap replied:
Ok thanks moved it here.

lzap commented:
Rebased.

ares commented:
perhaps another rebase would fix the foreman failures, seems no conflicts so that should be easy

lzap commented:
Rebased.

tbrisker commented:
tests are failing.

lzap commented:
Found the issue due to refactoring of the method, fixed, hopefully tests are green now. Rebased. Assigned to you @tbrisker if you don't mind, this one is straightforward (if its green).

tbrisker reviewed (approved):
Thanks @lzap !
